plugins/big-number/src/BigNumber/transformProps.ts

```diff
@@ -19,25 +19,58 @@
 import * as color from 'd3-color';
 import { getNumberFormatter, NumberFormats } from '@superset-ui/number-format';
 import { ChartProps } from '@superset-ui/chart';
-import getTimeFormatterForGranularity from '../utils/getTimeFormatterForGranularity';
+import getTimeFormatterForGranularity, {
+  TimeGranularity,
+} from '../utils/getTimeFormatterForGranularity';
 
 const TIME_COLUMN = '__timestamp';
 const formatPercentChange = getNumberFormatter(NumberFormats.PERCENT_SIGNED_1_POINT);
 
+export interface DatasourceMetric {
+  label: string;
+  // eslint-disable-next-line camelcase
+  metric_name?: string;
+  d3format?: string;
+}
+
 // we trust both the x (time) and y (big number) to be numeric
-type BigNumberDatum = {
-  [TIME_COLUMN]: number;
+export interface BigNumberDatum {
   [key: string]: number | null;
+}
+
+export type BigNumberData = BigNumberDatum[];
+
+export type BigNumberFormData = {
+  colorPicker?: {
+    r: number;
+    g: number;
+    b: number;
+  };
+  metric?:
+    | {
+        label: string;
+      }
+    | string;
+  compareLag?: string | number;
+  yAxisFormat?: string;
+  timeGrainSqla?: TimeGranularity;
+} & ChartProps['formData'];
```

Comment:
Thought it was more semantically correct to include it. Can't probably drop since we've been exhaustive listing all the fields.

Reply:
two questions:
- should it be `Partial<ChartProps['formData']>`? or all of them?
- are there duplicate keys in `ChartProps['formData']`? I'm not sure order of override for TS, just wanted to make sure if there are duplicates that the `BigNumberFormData` values take precedent.

Reply:
<strike>Actually, it'd be easier to just do this because some shared properties are indeed useful to keep in ChartProps and it doesn't make sense to exhaustively list them for every chart type.</strike>

Strike again. `formData` and `queryData` are basically `PlainObject`, which accepts any key with any values. `Partial` or not it doesn't make any difference, at least for now. I was thinking `& ChartProps`.

@williaster thanks for the reminder about overrides. I should have placed `ChartProps['formData']` at the front.

Reply:
I think I'm going to avoid `Partial` as it makes all properties optional, which is probably not what we want in this case.